src/synapse.py

```diff
@@ -30,13 +30,15 @@
 class Synapse:
     """A class representing the Synapse application."""
 
-    def __init__(self, charm_state: CharmState):
+    def __init__(self, charm_state: CharmState, database_data: typing.Optional[typing.Dict]):
```

Comment:
At this stage should the database data be in the state?

Reply:
Yes, changed.